Amplify/DevMenu/View/EnvironmentInfoDetailView.swift

```diff
@@ -0,0 +1,159 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import SwiftUI
+
+/// Detail view showing environment information
+@available(iOS 13.0.0, *)
+struct EnvironmentInfoDetailView: View {
+    private let screenTitle  = "Environment Information"
+    private let amplifyEnvSectionTitle  = "Amplify Environment Information"
+    private let devEnvSectionTitle  = "Developer Environment Information"
+    private let notAvailable = "Not Available"
+
+    // Key descriptions for environment information
+    private let nodeJsVersionDesc = "Node.js version"
+    private let npmVersionDesc = "npm version"
+    private let amplifyCLIVersionDesc = "Amplify CLI version"
+    private let podVersionDesc = "CocoaPods version"
+    private let xcodeVersionDesc = "Xcode version"
+    private let osVersionDesc = "macOS version"
+
+    /// Lists containing items belonging to two sections : Amplify and Developer Environment Informatoin
+    private var amplifyEnvSectionItems = [EnvironmentInfoItem]()
+    private var devEnvSectionItems = [EnvironmentInfoItem]()
+
+    init() {
+        // TODO : Read environment information from json file and
+        // call populateDevEnvSectionItems()
+        populateAmplifyEnvSectionItems()
+    }
+
+    var body: some View {
+        SwiftUI.List {
+            Section(header: Text(amplifyEnvSectionTitle), footer: EmptyView()) {
+                if amplifyEnvSectionItems.isEmpty {
+                    NoItemView()
+                } else {
+                    ForEach(amplifyEnvSectionItems) { listItem in
+                        EnvironmentInfoRow(rowItem: listItem)
+                    }
+                }
+            }
+            Section(header: Text(devEnvSectionTitle), footer: EmptyView()) {
+                if devEnvSectionItems.isEmpty {
+                    NoItemView()
+                } else {
+                    ForEach(devEnvSectionItems) { listItem in
+                        EnvironmentInfoRow(rowItem: listItem)
+                    }
+                }
+            }
+        }
+        .navigationBarTitle(screenTitle)
+        .listStyle(GroupedListStyle())
+    }
+
+    private mutating func populateDevEnvSectionItems(devEnvInfo: DevEnvironmentInfo) {
+        devEnvSectionItems.append(EnvironmentInfoItem(
+            key: nodeJsVersionDesc,
+            value: devEnvInfo.nodejsVersion.isEmpty ? notAvailable : devEnvInfo.nodejsVersion))
```

Comment:
maybe `value` could be nullable and then the `notAvailable` logic happens inside the `InfoItem` view